packages/x-data-grid/src/components/columnSelection/GridCellCheckboxRenderer.tsx

```diff
@@ -50,6 +54,29 @@ const GridCellCheckboxForwardRef = React.forwardRef<HTMLInputElement, GridRender
     const classes = useUtilityClasses(ownerState);
     const checkboxElement = React.useRef<HTMLElement>(null);
 
+    const rowSelectionLookup = useGridSelector(apiRef, selectedIdsLookupSelector);
+    const filteredRowsLookup = useGridSelector(apiRef, gridFilteredRowsLookupSelector);
+    const children: GridRowId[] = React.useMemo(() => {
+      if (rowNode.type === 'group') {
+        // @ts-expect-error Access `GridProApi` method
+        return apiRef.current.getRowGroupChildren({
+          groupId: id,
+          skipAutoGeneratedRows: false,
+          applySorting: true,
```

Comment:
Not sure if I understand correctly what it does, but is `applySorting` useful here?

Reply:
It may sound weird, but in practice, using the `sortedRowIds` (or `applySorting`) to get row group children is faster than not passing `applySorting`.
The reason seems to be the loops being faster than recursive logic.
In my tests, it was almost 40% faster for the same data size.

Reply:
If that option is faster, why does it exist? Haven't look at any of that code yet I think.

Reply:
> If that option is faster, why does it exist

Not sure tbh, maybe the perf benefits were not considered while implementing this [function](https://github.com/mui/mui-x/blob/cd68580bd2c80a44960952ce8fbdddf7c2ccbd89/packages/x-data-grid/src/hooks/features/rows/useGridRows.ts#L302-L339).
I'm inclining to use a selector instead of this method though as I spotted a possible reactivity bug related to this feature.

Reply:
[Added](https://github.com/mui/mui-x/pull/13757/commits/ff2bea17314054e562f3186abc20be0d3efffabd) selectors, which seem to fix the reactivity bug as well as improve the performance further ~150-300ms (down to ~1.4s in development mode) on 1M rows data set.